test/functional-portable/dynamicrp/noncloud/resources/dynamicrp_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/radius-project/radius/test"
+	"github.com/radius-project/radius/test/radcli"
+	"github.com/radius-project/radius/test/rp"
+	"github.com/radius-project/radius/test/step"
+	"github.com/radius-project/radius/test/testutil"
+	"github.com/radius-project/radius/test/validation"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_DynamicRP_Recipe(t *testing.T) {
+	template := "testdata/usertypealpha-recipe.bicep"
+	name := "usertypealpha-recipe-app"
+	appNamespace := "default-usertypealpha-recipe"
+	resourceTypeName := "Test.Resources/userTypeAlpha"
+	filepath := "testdata/usertypealpha.yaml"
+	options := rp.NewRPTestOptions(t)
+	cli := radcli.NewCLI(t, options.ConfigFilePath)
+
+	test := rp.NewRPTest(t, name, []rp.TestStep{
+		{
+			// The first step in this test is to create/register a resource type using the CLI.
+			Executor: step.NewFuncExecutor(func(ctx context.Context, t *testing.T, options test.TestOptions) {
+				output, err := cli.ResourceProviderCreate(ctx, filepath)
+				require.NoError(t, err)
+				require.Contains(t, output, "Creating resource type "+resourceTypeName)
```

Comment:
What's the benefit of validating this output string? I'd recommend running resource type show and validating that it returns expected resource type in output instead, perhaps in `PostStepVerify` step?